### PR TITLE
fix service bug in windows

### LINKED ISSUE #19: demon start failed in windows platform
# windows 平台 daemon 命令运行报错

windows 平台下 `mg daemon start` 运行报错。
我觉得应该是 [mg daemon start](https://github.com/changkun/midgard/blob/main/cmd/daemon.go#L56) 没有调用 s.Run 方法的原因。

## 复现步骤

1. windows 平台下执行 `make` 打包应用。
2. 执行 `./mg daemon install` 注册服务。
3. 执行 `./mg daemon start` 运行服务。
4. 一段时间后控制台报错 midgard: failed to start, err: The service did not respond to the start or control request in a timely fashion.

## 解释

个人猜测，windows 平台下服务注册与启动流程应该是这样的：

1. Install 方法将 `path/to/your/service run` 注册到系统服务中。
2. Start 方法通知系统后台执行 `path/to/your/service run` 命令。
3. 命令被系统调用后。调用 Run 方法通知系统与 Handler 交互。
4. Handler 通过 Execute 方法与系统交互。在其他协程中运行业务逻辑。

而当前的 `mg daemon run` 命令直接在当前协程运行了业务逻辑，并且没使用 s.Run 方法。
导致 Handler 无法与系统进行交互。最终超时报错。

### PR DESCRIPTION
**Pull request by kvii**
Fixes #19

### CODE REVIEW

File: cmd/daemon.go [hunk 1]
```diff
@@ -54,8 +54,7 @@ var daemonCmd = &cobra.Command{
 		case "stop":
 			err = s.Stop()
 		case "run":
-			m := daemon.NewDaemon()
-			m.Serve()
+			err = s.Run(daemon.NewDaemon().Run(context.Background()))
```
Comment (changkun) [inline cmd/daemon.go:57]: We should check if the error is nil or not. If not, then we need to log it out.

Comment (changkun) [reply cmd/daemon.go:57]: Also, this line is a little bit unintuitive to read. Could we do a little bit of separation here?

```
m := daemon.NewDaemon()
onStart, onStop := m.Run(context.Background())
err = s.Run(onStart, onStop)
...
```

File: internal/service/service_windows.go [hunk 2]
```diff
@@ -124,11 +126,32 @@ func (ws *windowsService) Run(onStart, onStop func() error) error {
 
 	ws.logger = elog
 
+	ok, err := svc.IsWindowsService()
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return ws.runNotSerVice(onStart, onStop)
+	}
```
Comment (changkun) [inline internal/service/service_windows.go:135]: I am not sure what's happening here. Could you add a comment explaining why we need to do this?

Comment (kvii) [reply internal/service/service_windows.go:135]: "service mode": mg is executed by windows as a service.
"normal mode": mg is executed by user. e.g. mg daemon run.
`svc.Run` can only be called in "service mode".

**Review by changkun — APPROVED** [review]
I looked high-level details and all changes seem reasonable to me.

**Pull request merged**